### ISSUE TITLE
Variable use like "${$a}" isn't reported as an error and is changed with no message when run with --fix

### ISSUE THREAD

**Issue by cbowman0** [rodjek/puppet-lint#655]
Sample init.pp:
```
#
#
#
class test  {

  $a = 'string'

  $b = "${$a}/current"
}
```

Running puppet-lint reports no errors and exits with status 0.

Running `puppet-lint --fix init.pp` modifies the file to the following but produces no output reporting changes:

```
#
#
#
class test  {

  $a = 'string'

  $b = "${a}/current"
}
```
